languages/python/oso/tests/data_filtering/test_basic.py

```diff
@@ -555,3 +556,147 @@ class Qux:
 
     result = oso.authorized_resources("gwen", "read", Qux)
     assert len(result) == 1
+
+
+def test_complex_isa(oso):
+    """Failing example from user @moea"""
+
+    policy = """
+        actor User {}
+
+        resource Group {
+            relations = {network: Network};
+            permissions = ["read"];
+            roles = ["host", "member"];
+
+            "read" if "member";
+            "member" if "host";
+        }
+
+        resource Network {
+          relations = {members: NetworkMember};
+          permissions = ["read"];
+          roles = ["member"];
+
+        }
+        resource NetworkMember {
+            relations = {user: User};
+        }
+
+        resource Article {
+          relations = {group: Group};
+          permissions = ["read"];
+        }
+
+        allow(_, "read", _n: Network);
+
+        has_permission(u: User, "read", a: Article{group}) if
+        a matches Article and
+            group matches Group and
+            has_permission(u, "read", group);
+
+        has_permission(u: User, "read", g: Group{network}) if
+        g matches Group and
+            network matches Network and
```

Comment:
seeing gabe's comment in slack, would be good to remove these `matches` if possible.

Reply:
✂️